geomesa-dynamo/geomesa-dynamo-core/src/main/scala/org/locationtech/geomesa/dynamo/core/DynamoGeoQuery.scala

```diff
@@ -0,0 +1,126 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.dynamo.core
+
+import com.vividsolutions.jts.geom.Envelope
+import org.geotools.data.simple.DelegateSimpleFeatureReader
+import org.geotools.data.{FeatureReader, Query}
+import org.geotools.feature.collection.DelegateSimpleFeatureIterator
+import org.geotools.filter.visitor.ExtractBoundsFilterVisitor
+import org.geotools.geometry.jts.ReferencedEnvelope
+import org.geotools.referencing.crs.DefaultGeographicCRS
+import org.joda.time.Interval
+import org.locationtech.geomesa.filter._
+import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType._
+import org.locationtech.sfcurve.IndexRange
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+import org.opengis.filter.Filter
+
+import scala.collection.GenTraversable
+import scala.collection.JavaConverters.asJavaIteratorConverter
+
+trait DynamoGeoQuery {
+
+  case class HashAndRangeQueryPlan(row: Int, lz3: Long, uz3: Long, contained: Boolean)
+
+  protected val WHOLE_WORLD = new ReferencedEnvelope(-180.0, 180.0, -90.0, 90.0, DefaultGeographicCRS.WGS84)
+
+  def getAllFeatures: Iterator[SimpleFeature]
+
+  def getAllFeatures(filter: Seq[Filter]): Iterator[SimpleFeature] = {
+    getAllFeatures.filter(f => filter.forall(_.evaluate(f)))
+  }
+
+  def executeGeoTimeQuery(query: Query, plans: GenTraversable[HashAndRangeQueryPlan]): GenTraversable[SimpleFeature]
```

Comment:
I would make the return type an Iterator (with closeable if appropriate) - that's what works best with all the geotools classes.